generator/lib/src/code_chunks.dart

```diff
@@ -33,45 +36,109 @@ class CodeChunks {
         model: model.getEntityByUid(${entity.id.uid}),
         getId: ($name inst) => inst.${propertyFieldName(entity.idProperty)},
         setId: ($name inst, int id) {inst.${propertyFieldName(entity.idProperty)} = id;},
-        reader: ($name inst) => {
-          ${entity.properties.map(propertyReader).join(",\n")}
-        },
-        writer: (Map<String, dynamic> members) {
-          final r = $name();
-          ${entity.properties.map(propertyWriter).join()}
-          return r;
-        }
+        objectToFB: ${objectToFB(entity)},
+        objectFromFB: ${objectFromFB(entity)}
       )
       """;
   }
 
-  static String typedDataImportIfNeeded(ModelInfo model) {
-    if (model.entities
-        .any((ModelEntity entity) => entity.properties.any(isTypedDataList))) {
-      return "import 'dart:typed_data';\n";
-    }
-    return '';
-  }
-
-  static bool isTypedDataList(ModelProperty property) {
-    return ['Int8List', 'Uint8List'].contains(property.dartFieldType);
-  }
-
   static String propertyFieldName(ModelProperty property) {
     return property.name;
   }
 
-  static String propertyReader(ModelProperty property) {
-    return "'${property.name}': inst.${propertyFieldName(property)}";
+  static int propertyFlatBuffersSlot(ModelProperty property) =>
+      property.id.id - 1;
+
+  static int propertyFlatBuffersvTableOffset(ModelProperty property) =>
+      4 + 2 * propertyFlatBuffersSlot(property);
```

Comment:
Where does this calculation come from?

Reply:
ehh, I don't have any docs reference. It's just something I've found out from the flatc generated code and use in all the language bindings. e.g. [c++](https://github.com/objectbox/objectbox-generator/blob/main/test/comparison/testdata/fbs/typeful/cpp/schema.obx.cpp.expected#L36)

Reply:
K, so I'll just believe this works then. 😁